library/src/updater.rs

```diff
@@ -352,7 +354,7 @@ where
 /// This may be changed any time `update()` or `start_update_thread()` are called.
 pub fn next_boot_patch() -> anyhow::Result<Option<PatchInfo>> {
     with_config(|config| {
-        let state =
+        let mut state =
```

Comment:
Why is this mut?  You might want to use some tricks to internal mutability in next_boot_patch() if it's mutating (I can look up how I did it in safe_wren).

Reply:
This updater state is mutable because it calls `self.patch_manager.next_boot_patch()`, which can mutate the patch manager that it owns. A little weird, I agree.

Reply:
I believe rust has ways to hide internal mutability through a Ref, but I'd have to look at my old code.

Reply:
RefCell is what I'm thinking of.

Reply:
Ah gotcha. It looks like RefCell is [unsafe](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html), so I'm inclined to stick to the odd-looking `mut state` for now unless you have strong objections.

Reply:
sgtm.